From 0cc335a411127feb05e16cbf2803058b2db1b405 Mon Sep 17 00:00:00 2001
From: Meng Mingming <mingming.meng@nxp.com>
Date: Thu, 27 Oct 2016 17:00:24 +0800
Subject: [PATCH 1/2] MGS-2343 [#ccc] XWLD: T3DStressTest_Wayland displays
 abnormally while using G2D compositor

It need add g2d_finish after repaint_region and before copy to framebuffer.
Because if using dual dpu cores, the tasks on each core will be executed
sequently, but it can't ensure one task splited on dual cores can be finished
synchronously by hardware.

Upstream-Status: Inappropriate [i.MX specific]

Date: Oct 27, 2016
Signed-off-by: Meng Mingming <mingming.meng@nxp.com>
---
 src/g2d-renderer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/g2d-renderer.c b/src/g2d-renderer.c
index 1009115..89c69bf 100644
--- a/src/g2d-renderer.c
+++ b/src/g2d-renderer.c
@@ -609,6 +609,7 @@ repaint_region(struct weston_view *ev, struct weston_output *output, struct g2d_
 			g2d_set_clipping(gr->handle, clipRect.left, clipRect.top, clipRect.right, clipRect.bottom);
 			g2d_blitSurface(gr->handle, &gs->g2d_surface, dstsurface, &srcRect, &dstrect);
 		}
+		g2d_finish(gr->handle);
 	}
 }
 
-- 
2.7.4

